curate_export/curate_api.py

```diff
@@ -0,0 +1,112 @@
+# curate_api.py
+import time
+import json
+from datetime import datetime
+import dependencies.requests
+from dependencies.sentry_sdk import capture_exception
+from translate_curate_json_node import TranslateCurateJsonNode
+from csv_from_json import CsvFromJson
+from save_csv import SaveCsv
+from create_standard_json import CreateStandardJson
+
+
+class CurateApi():
+    """ This performs all Curate API processing """
+    def __init__(self, config, event, time_to_break):
+        self.config = config
+        self.event = event
+        self.curate_header = {"X-Api-Token": self.config["curate-token"]}
+        self.start_time = time.time()
+        self.time_to_break = time_to_break
+        self.translate_curate_json_node_class = TranslateCurateJsonNode(config, event)
+        self.save_standard_json_locally = event.get("local", False)
+        self.save_curate_json_locally = event.get("local", False)
+        self.create_standard_json_class = CreateStandardJson(config, event)
+        self.csv_from_json_class = CsvFromJson(self.config.get("csv-field-names", []))
+        self.save_csv_class = SaveCsv(config, event)
+
+    def get_curate_items(self, ids):
+        """ Given a list of ids, process each one that corresponds to a Curate item """
+        aborted_processing = False
+        for id in list(ids):  # iterate over a copy of the list, so we can remove items from the original list
+            if "und:" in id:
+                curate_id = id.replace("und:", "")  # strip namespace
+                self.get_curate_item(curate_id)
+                ids.remove(id)
+            if datetime.now() > self.time_to_break and len(ids) > 0:
+                aborted_processing = True
+                break
+        return not(aborted_processing)
+
+    def get_curate_item(self, id):
+        """ Get json metadata for a curate item given an item id
+            Note: query is of the form: curate-server-base-url + "/api/items/<pid>" """
+        standard_json = {}
+        url = self.config["curate-server-base-url"] + "/api/items/" + id
+        curate_json = self._get_json_given_url(url)
+        members = []
+        if "membersUrl" in curate_json:
+            members = self._get_members_list(curate_json['membersUrl'], id)
+            members = self._get_members_details(members)
+            curate_json["members"] = members
+        standard_json = self.translate_curate_json_node_class.build_json_from_curate_json(curate_json, "root")
+        standard_json = self.create_standard_json_class.build_standard_json_tree(standard_json, members)
+        csv_string = self.csv_from_json_class.return_csv_from_json(standard_json)
+        self.save_csv_class.save_csv_file(id, csv_string)
```

Comment:
I think these would be better as local variables in get_curate_item instead of class variables. You could then remove the 4 class variables, but it's up to you if you prefer the current setup.

```suggestion
        standard_json = TranslateCurateJsonNode(self.config, self.event).build_json_from_curate_json(curate_json, "root")
        standard_json = CreateStandardJson(self.config, self.event).build_standard_json_tree(standard_json, members)
        csv_string = CsvFromJson(self.config.get("csv-field-names", [])).return_csv_from_json(standard_json)
        SaveCsv(self.config, self.event).save_csv_file(id, csv_string)

Reply:
Ryan, Thanks for the suggestions.  I originally had something similar to what you suggested, but because I made the changes to be more efficient.  For example, the TranslateCurateJsonNode reads a file - cureate_to_json_translation_control_file.json - into memory when it instantiates.  Because of that, I thought it would be best to just instantiate it once.  What would be best practices in this situation?  Thanks for your suggestions.

Reply:
@SteveMattison makes sense. looks good as is then